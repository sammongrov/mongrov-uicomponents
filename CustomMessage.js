import React, { Component } from 'react';
import { TouchableOpacity, View, Text, Clipboard, Alert } from 'react-native';
import { Colors } from '@ui/theme_default';
import { styles } from 'react-native-theme';
import { MessageText } from 'react-native-gifted-chat';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import { FeatherIcon } from '@ui/components';
import FastImage from 'react-native-fast-image';
import {DBManager} from 'app-module';

export default class CustomMessage extends Component {
  state = {
    showMessageActions: false,
    messageActionsFor: '',
    repliesCounter: 0,
    msgHasLikes: false,
    msgHasReplies: false,
    bubblePress: false,
  };

  componentWillMount() {
    const { currentMessage } = this.props;
    const numOfReplies = this.getNumOfReplies(currentMessage._id, currentMessage.group);
    if (currentMessage.likes > 0) {
      this.setState({ msgHasLikes: true });
    }
    if (numOfReplies > 0) {
      this.setState({ msgHasReplies: true, repliesCounter: numOfReplies });
    }
  }

  /* eslint-disable */
  componentDidUpdate(prevProps, prevState) {
    // console.log(`MESSAGE ${this.props.currentMessage.text} is updated.`);
    const { currentMessage } = this.props;
    const numOfReplies = this.getNumOfReplies(currentMessage._id, currentMessage.group);
    if (prevProps.currentMessage.likes !== currentMessage.likes) {
      this.setState({ msgHasLikes: true });
    }
    if (prevState.repliesCounter !== numOfReplies) {
      this.setState({ msgHasReplies: true, repliesCounter: numOfReplies });
    }
  }
  /* eslint-enable */

  handleLike = (messageId) => {
    const { likeMessage } = this.props;
    likeMessage(messageId);
    this.setState({ showMessageActions: false, messageActionsFor: '' });
  };

  handleReply = (message) => {
    const { replyMessage } = this.props;
    replyMessage(message);
    this.setState({ showMessageActions: false, messageActionsFor: '' });
  };

  handleCopy = (text) => {
    Clipboard.setString(text);
    this.setState({ showMessageActions: false, messageActionsFor: '' });
  };

  handleMessageInfo = (message) => {
    const { messageInfo } = this.props;
    messageInfo(message);
    this.setState({ showMessageActions: false, messageActionsFor: '' });
  };

  handleDelete = (message) => {
    const { deleteMessage } = this.props;
    Alert.alert(
      'Delete',
      'Do you want to delete message?',
      [
        { text: 'No', onPress: () => {}, style: 'cancel' },
        {
          text: 'Yes',
          onPress: () => {
            deleteMessage(message._id, message.user._id);
            this.setState({ showMessageActions: false, messageActionsFor: '' });
          },
        },
      ],
      { cancelable: false },
    );
  };

  getNumOfReplies = (messageId, groupId) => {
    const numOfReplies = DBManager.group.getNumOfMessageReplies(messageId, groupId);
    return numOfReplies;
  };

  renderReply = (props) => {
    const { replyMessageId } = props.currentMessage;
    const replyMessageObj = DBManager.group.findMessageById(replyMessageId);
    // console.log('REPLY MESSAGE ID, OBJ', replyMessageId, replyMessageObj);

    if (replyMessageId && replyMessageObj) {
      const { user, text, image } = replyMessageObj;
      const sideColor = props.position === 'left' ? Colors.TEXT_DARK : Colors.TEXT_REPLY_RIGHT;
      return (
        <View style={styles.threadedMessageContainer}>
          <View style={styles.threadedMessageView}>
            <Text
              style={[styles.threadedMessageText, { color: sideColor }]}
              numberOfLines={1}
            >{`${user.name || user.username}:`}</Text>
            <Text style={{ color: sideColor }}>{text}</Text>
          </View>
          {image && (
            <TouchableOpacity
              onPress={() => {
                if (Actions.currentScene === 'ChatRoomScene') {
                  // this.setState({ actionsMenu: false });
                  Actions.ViewImage({
                    imageUrl: image,
                    goBack: () => Actions.pop,
                  });
                }
              }}
              style={styles.padding3}
            >
              <FastImage
                style={styles.threadedMessageImage}
                source={{
                  uri: image,
                  priority: FastImage.priority.high,
                }}
              />
            </TouchableOpacity>
          )}
        </View>
      );
    }
  };

  renderMessageActions = (props) => {
    const { position, currentMessage, canDelete, isSameUser } = props;
    const { messageActionsFor, repliesCounter, msgHasLikes, msgHasReplies } = this.state;
    // const repliesCounter = this.getNumOfReplies(currentMessage._id, currentMessage.group);
    // console.log('SHOW MESSAGE ACTIONS', messageActionsFor);
    // console.log('Replies counter', repliesCounter, 'for', currentMessage._id);

    const iconColor = position === 'left' ? '#696969' : Colors.ICON_WHITE;
    const borderColor = position === 'left' ? '#dfdfdf' : '#70B1FF';
    if (messageActionsFor !== currentMessage._id && (msgHasLikes || msgHasReplies)) {
      return (
        <View
          style={[
            styles.rowDirection,
            {
              borderTopColor: borderColor,
              borderTopWidth: 1,
              justifyContent: 'flex-start',
              alignItems: 'center',
              marginLeft: 10,
              marginRight: 8,
              paddingVertical: 10,
            },
          ]}
        >
          {msgHasLikes && (
            <TouchableOpacity
              style={[
                styles.rowDirection,
                {
                  alignItems: 'center',
                  justifyContent: 'flex-start',
                  marginRight: 20,
                },
              ]}
              onPress={() => this.handleLike(currentMessage._id)}
            >
              <FeatherIcon name="thumbs-up" color={iconColor} size={20} />
              <Text style={{ color: iconColor, fontSize: 16, paddingLeft: 5 }}>
                {currentMessage.likes}
              </Text>
            </TouchableOpacity>
          )}
          {msgHasReplies && (
            <TouchableOpacity
              style={[
                styles.rowDirection,
                {
                  alignItems: 'center',
                  justifyContent: 'flex-start',
                  marginRight: 20,
                },
              ]}
              onPress={() => this.handleReply(currentMessage)}
            >
              <FeatherIcon name="corner-up-left" color={iconColor} size={20} />
              <Text style={{ color: iconColor, paddingLeft: 5, fontSize: 16 }}>
                {repliesCounter}
              </Text>
            </TouchableOpacity>
          )}
        </View>
      );
    }
    if (messageActionsFor === currentMessage._id) {
      return (
        <View
          style={[
            styles.rowDirection,
            {
              borderTopColor: borderColor,
              borderTopWidth: 1,
              justifyContent: 'space-between',
              alignItems: 'center',
              marginLeft: 10,
              marginRight: 8,
              paddingVertical: 10,
            },
          ]}
        >
          <TouchableOpacity
            style={[
              styles.rowDirection,
              {
                alignItems: 'center',
              },
            ]}
            onPress={() => this.handleLike(currentMessage._id)}
          >
            <FeatherIcon name="thumbs-up" color={iconColor} size={20} />
            <Text style={{ color: iconColor, fontSize: 16, paddingLeft: 5 }}>
              {currentMessage.likes}
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={[
              styles.rowDirection,
              {
                alignItems: 'center',
              },
            ]}
            onPress={() => this.handleReply(currentMessage)}
          >
            <FeatherIcon name="corner-up-left" color={iconColor} size={20} />
            <Text style={{ color: iconColor, paddingLeft: 5, fontSize: 16 }}>{repliesCounter}</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.handleCopy(currentMessage.text)}>
            <FeatherIcon name="copy" color={iconColor} size={20} />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => this.handleMessageInfo(currentMessage)}>
            <FeatherIcon name="alert-circle" color={iconColor} size={20} />
          </TouchableOpacity>
          {canDelete &&
            isSameUser && (
              <TouchableOpacity onPress={() => this.handleDelete(currentMessage)}>
                <FeatherIcon name="trash-2" color={iconColor} size={20} />
              </TouchableOpacity>
            )}
        </View>
      );
    }
  };

  render() {
    const { currentMessage } = this.props;
    const { showMessageActions, messageActionsFor, bubblePress } = this.state;
    // console.log('STATE MESSAGE ACTIONS', showMessageActions);
    // console.log('MESSAGE ACTIONS ID', messageActionsFor);
    return (
      <View style={bubblePress ? { minWidth: 250 } : null}>
        <TouchableOpacity
          onPress={() => {
            if (!showMessageActions) {
              const repliesCounter = this.getNumOfReplies(currentMessage._id, currentMessage.group);
              this.setState({
                repliesCounter,
                bubblePress: !bubblePress,
              });
            }
            this.setState({
              showMessageActions: !showMessageActions,
              messageActionsFor: messageActionsFor === '' ? currentMessage._id : '',
              bubblePress: !bubblePress,
            });
          }}
        >
          {this.renderReply(this.props)}
          <MessageText {...this.props} />
          {this.renderMessageActions(this.props)}
        </TouchableOpacity>
      </View>
    );
  }
}

CustomMessage.propTypes = {
  currentMessage: PropTypes.object,
  position: PropTypes.string,
  canDelete: PropTypes.bool,
  isSameUser: PropTypes.bool,
  replyMessage: PropTypes.func,
  messageInfo: PropTypes.func,
  deleteMessage: PropTypes.func,
  likeMessage: PropTypes.func,
};

CustomMessage.defaultProps = {
  currentMessage: {},
  position: '',
  canDelete: false,
  isSameUser: false,
  replyMessage: () => {},
  messageInfo: () => {},
  deleteMessage: () => {},
  likeMessage: () => {},
};
