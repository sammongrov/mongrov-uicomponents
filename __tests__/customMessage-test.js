import React from 'react';
import { Clipboard, Alert } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import {DBManager} from 'app-module';

import CustomMessage from '../CustomMessage';

configure({ adapter: new Adapter() });

jest.mock('react-native-router-flux', () => ({
  Actions: {
    currentScene: '',
    ViewImage: jest.fn(),
    pop: jest.fn(),
  },
}));

jest.mock('../../app/DBManager', () => {
  const dbManager = {
    group: {
      findMessageById: jest.fn(() => ({
        user: { _id: 'user123', name: '', username: 'test-username' }, // 'test-user'
        text: 'test-text1',
        image: 'http://test-images/test-image1.png',
      })),
      getNumOfMessageReplies: jest.fn(() => 7),
    },
  };
  return dbManager;
});

jest.mock('Alert', () => ({
  alert: jest.fn((str1, str2, arr) => {
    arr[0].onPress();
  }),
}));

const currentMessage = {
  _id: '123987',
  user: {
    _id: '6968745',
    name: 'test-user',
  },
  replyMessageId: '11111',
  group: 'new-group',
  type: 1,
  isReply: true,
  status: 1,
  likes: 5,
  text: 'test-text',
  createdAt: '2017-11-24T19:21:42.183Z',
  updatedAt: '2017-11-24T19:21:42.183Z',
};

let position = 'left';
let canDelete = false;
let isSameUser = false;

beforeEach(() => {
  jest.resetModules();
});

it('first TO to be called', () => {
  currentMessage.likes = 0;
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  // console.log(customMessage.debug());
  const TO = customMessage.find('TouchableOpacity').first();
  TO.props().onPress();

  expect(customMessage.state().repliesCounter).toBe(7);
  expect(customMessage.state().bubblePress).toBe(true);
  expect(customMessage.state().showMessageActions).toBe(true);
  expect(customMessage.state().messageActionsFor).toBe('123987');
});

it('first TO to be called', () => {
  //
  currentMessage.likes = 45;
  //
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  customMessage.setState({
    showMessageActions: true,
    bubblePress: false,
    messageActionsFor: '123456',
  });
  const instance = customMessage.instance();
  instance.getNumOfReplies = jest.fn(() => 14);
  const TO = customMessage.find('TouchableOpacity').first();
  TO.props().onPress();

  expect(customMessage.state().repliesCounter).toBe(14);
  expect(customMessage.state().bubblePress).toBe(true);
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('renderReply TO is called with else case', () => {
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  const TO = customMessage
    .find('FastImage')
    .first()
    .parent();
  TO.props().onPress();

  expect(Actions.currentScene).toEqual('');
  expect(Actions.ViewImage).not.toBeCalled();
});

it('renderReply TO is called', () => {
  Actions.currentScene = 'ChatRoomScene';
  Actions.pop.mockClear();
  Actions.ViewImage = jest.fn((props) => {
    props.goBack();
  });
  position = 'right';
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  const TO = customMessage
    .find('FastImage')
    .first()
    .parent();
  TO.props().onPress();

  expect(Actions.currentScene).toEqual('ChatRoomScene');
  expect(Actions.ViewImage).toBeCalled();
});

it('renderReply else case, renderMessageActions-msgHasReplies TO is called ', () => {
  currentMessage.replyMessageId = '';
  currentMessage.likes = 0;

  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );

  // console.log(customMessage.debug());
  expect(customMessage.state().messageActionsFor).toBe('');
  expect(customMessage.state().showMessageActions).toBe(false);

  const instance = customMessage.instance();
  instance.handleReply = jest.fn();
  const TO = customMessage.find({ name: 'corner-up-left' }).parent();
  TO.props().onPress();

  expect(instance.handleReply).toBeCalledWith(currentMessage);
});

it('renderMessageActions-msgHasLikes TO is called ', () => {
  currentMessage.replyMessageId = '11111';
  currentMessage.likes = 12;

  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );

  // console.log(customMessage.debug());
  expect(customMessage.state().messageActionsFor).toBe('');
  expect(customMessage.state().showMessageActions).toBe(false);

  const instance = customMessage.instance();
  instance.handleLike = jest.fn();
  const TO = customMessage.find({ name: 'thumbs-up' }).parent();
  TO.props().onPress();

  expect(instance.handleLike).toBeCalledWith(currentMessage._id);
});

it('renderMessageActions 2-if case, all TO is called', () => {
  DbManager.group.getNumOfMessageReplies = jest.fn(() => 0);
  canDelete = true;
  isSameUser = true;

  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  customMessage.setState({ messageActionsFor: '123987' });
  // console.log(customMessage.debug());
  const inst = customMessage.instance();
  inst.handleLike = jest.fn();
  inst.handleReply = jest.fn();
  inst.handleCopy = jest.fn();
  inst.handleMessageInfo = jest.fn();
  inst.handleDelete = jest.fn();

  const TO1 = customMessage.find({ name: 'thumbs-up' }).parent();
  TO1.props().onPress();
  expect(inst.handleLike).toBeCalledWith(currentMessage._id);

  const TO2 = customMessage.find({ name: 'corner-up-left' }).parent();
  TO2.props().onPress();
  expect(inst.handleReply).toBeCalledWith(currentMessage);

  const TO3 = customMessage.find({ name: 'copy' }).parent();
  TO3.props().onPress();
  expect(inst.handleCopy).toBeCalledWith(currentMessage.text);

  const TO4 = customMessage.find({ name: 'alert-circle' }).parent();
  TO4.props().onPress();
  expect(inst.handleMessageInfo).toBeCalledWith(currentMessage);

  const TO5 = customMessage.find({ name: 'trash-2' }).parent();
  TO5.props().onPress();
  expect(inst.handleDelete).toBeCalledWith(currentMessage);
});

it('handleLike', () => {
  const likeMessage = jest.fn();
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
      likeMessage={likeMessage}
    />,
  );

  const inst = customMessage.instance();
  const messageId = '555';
  inst.handleLike(messageId);

  expect(likeMessage).toBeCalled();
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleReply', () => {
  const replyMessage = jest.fn();
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
      replyMessage={replyMessage}
    />,
  );

  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleReply(message);

  expect(replyMessage).toBeCalled();
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleMessageInfo', () => {
  const messageInfo = jest.fn();
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
      messageInfo={messageInfo}
    />,
  );
  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleMessageInfo(message);

  expect(messageInfo).toBeCalled();
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleCopy', () => {
  jest.mock('Clipboard', () => ({
    setString: jest.fn(),
  }));

  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  const text = 'clipBoard-test-text';
  const inst = customMessage.instance();
  inst.handleCopy(text);

  expect(Clipboard.setString).toBeCalled();
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleDelete, No button', () => {
  const deleteMessage = jest.fn();
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
      deleteMessage={deleteMessage}
    />,
  );
  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleDelete(message);

  expect(Alert.alert).toBeCalled();
});

it('handleDelete, Yes button', () => {
  jest.mock('Alert', () => ({
    alert: jest.fn((str1, str2, arr) => {
      arr[1].onPress();
    }),
  }));

  const deleteMessage = jest.fn();
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
      deleteMessage={deleteMessage}
    />,
  );
  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleDelete(message);

  expect(Alert.alert).toBeCalled();
  expect(deleteMessage).toBeCalled();
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleLike, likeMessage defaultProps', () => {
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );

  const inst = customMessage.instance();
  const messageId = '555';
  inst.handleLike(messageId);

  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleReply, replyMessage defaultProps', () => {
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );

  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleReply(message);

  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleMessageInfo, messageInfo defaultProps', () => {
  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleMessageInfo(message);

  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('handleDelete, Yes button, deleteMessage defaultProps', () => {
  jest.mock('Alert', () => ({
    alert: jest.fn((str1, str2, arr) => {
      arr[1].onPress();
    }),
  }));

  const customMessage = shallow(
    <CustomMessage
      currentMessage={currentMessage}
      position={position}
      canDelete={canDelete}
      isSameUser={isSameUser}
    />,
  );
  const message = currentMessage;
  const inst = customMessage.instance();
  inst.handleDelete(message);

  expect(Alert.alert).toBeCalled();
  expect(customMessage.state().showMessageActions).toBe(false);
  expect(customMessage.state().messageActionsFor).toBe('');
});

it('componentDidUpdate', () => {
  const customMessage = shallow(<CustomMessage currentMessage={currentMessage} />);
  currentMessage.likes = 25;
  customMessage.setProps({ currentMessage });

  expect(customMessage.state().msgHasLikes).toBe(true);
});

it('if line 202 else value, and line 147 else value', () => {
  currentMessage.likes = 0;
  DbManager.group.getNumOfMessageReplies = jest.fn(() => 0);
  const tree = renderer
    .create(
      <CustomMessage
        currentMessage={currentMessage}
        position={position}
        canDelete={canDelete}
        isSameUser={isSameUser}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
