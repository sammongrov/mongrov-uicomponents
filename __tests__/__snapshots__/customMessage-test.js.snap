// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if line 202 else value, and line 147 else value 1`] = `
<View
  style={null}
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              undefined,
              Object {
                "color": "#fff",
              },
            ]
          }
        >
          test-username:
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          test-text1
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "http://test-images/test-image1.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        childrenProps={Object {}}
        ellipsizeMode="tail"
        parse={
          Array [
            Object {
              "onPress": [Function],
              "style": Object {
                "color": "white",
                "textDecorationLine": "underline",
              },
              "type": "url",
            },
            Object {
              "onPress": [Function],
              "style": Object {
                "color": "white",
                "textDecorationLine": "underline",
              },
              "type": "phone",
            },
            Object {
              "onPress": [Function],
              "style": Object {
                "color": "white",
                "textDecorationLine": "underline",
              },
              "type": "email",
            },
          ]
        }
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "lineHeight": 20,
              "marginBottom": 5,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 5,
            },
            undefined,
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          test-text
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
